### ISSUE TITLE
e2e guestbook.sh flakes in creating loadbalancer

### ISSUE THREAD

**Issue by bprashanth** [GoogleCloudPlatform/kubernetes#5268]
Running e2e on my localsetup with a clean checkout, guestbook.sh flakes a lot more than any other test because of a timeout in creating the loadbalancer.

@rsokolowski can we remove the loadbalancer from guestbook.sh till we've figured out the cause of the timeouts? it would also help with confusion like https://github.com/GoogleCloudPlatform/kubernetes/issues/4464, because a loadbalancer is incompatible with localhost setup. Maybe just a separate lb test in e2e?

**Comment by goltermann** [comment]
This might be the only test currently testing the load balancer.  We should break this out into another test.  Perhaps in the short term we extend the timeout?

**Comment by bprashanth** [comment]
that would work if it really is a timeout and not a straight-up failure. you should be able to trigger the condition if you setup an e2e cluster and try guestbook.sh against it, no need to run the entire e2e suite (ie -build -pushup; ./guestbook.sh). You might also need https://github.com/GoogleCloudPlatform/kubernetes/pull/5262 to properly cleanup forwarding-rules.

**Comment by erictune** [comment]
seeing this issue too.
Failed 5 times.
Also, fails to cleanup properly for me.

At least call it a loadbalancer test, rather than guestbook.sh

**Comment by piosz** [comment]
I noticed the same problem #5431 and fixed it temporary by increasing timeout #5452. The goal is to create ELB asynchronously and @brendandburns is working on it #5180.

ELB is the only way to connect with application for users and it should be included in guestbook example - the most popular kubernetes example. And the test is testing actually guestbook example so it should use the same pod/service specifications.

Also I ported the test to ginkgo #5045 and I'm about to remove guestbook.sh e2e test #5536. The test in go no more uses ELB (but still create a one), so we probably need a separate e2e test that will test ELB, so creating a separate issue.

Last but not least the test is working significantly better now:)
